src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -259,7 +259,9 @@ public function hydrateValue(mixed $value, LivePropMetadata $propMetadata, objec
                 throw new \LogicException(sprintf('The "%s::%s" object should be hydrated with the Serializer, but no type could be guessed.', $parentObject::class, $propMetadata->getName()));
             }
 
-            return $this->serializer->denormalize($value, $propMetadata->getType(), 'json', $propMetadata->serializationContext());
+            $type = $propMetadata->collectionValueType() ? $propMetadata->collectionValueType()->getClassName().'[]' : $propMetadata->getType();
```

Comment:
Do we need to also check for `Type::BUILTIN_TYPE_OBJECT === $propMetadata->collectionValueType()->getBuiltinType()` like we do on line 267? Basically, what if this is typed as `string[]`?

Reply:
Yes absolutely, I missed those cases. Next commit fixes that, and also ensures the built-in collection type is scalar (if used). Otherwise the "not guessable type" exception is thrown